test/vue-directives.test.ts

```diff
@@ -0,0 +1,479 @@
+  },
+})
+
+const allDirectives = compileTemplate({
+  id: 'template.vue',
+  filename: 'template.vue',
+  source: `
+    <div v-awesome-directive v-custom-directive v-mixed-directive @click="foo"></div>
+  `,
+  compilerOptions: {
+    hoistStatic: false,
+  },
+})
+
+function replaceRoot(code: string) {
+  return code.replaceAll(process.cwd().replace(/\\/g, '/'), '<root>')
+}
+
+describe('vue-directives', () => {
+  describe('single default directive', () => {
+    const ctx = createUnimport({
+      presets: [{
+        from: '/directives/awesome-directive.ts',
+        imports: [{
+          name: 'default',
+          meta: {
+            vueDirective: true,
+          },
+        }],
+      }, {
+        from: '/directives/named-directive.ts',
+        imports: [{
+          name: 'NamedDirective',
+          meta: {
+            vueDirective: true,
+          },
+        }],
+      }],
+      addons: {
+        vueDirectives: true,
+      },
+    })
+    it('inject', async () => {
+      expect(replaceRoot(defaultDirective.code)).toMatchInlineSnapshot(`
+        "import { resolveDirective as _resolveDirective, withDirectives as _withDirectives, openBlock as _openBlock, createElementBlock as _createElementBlock } from "vue"
+
+        export function render(_ctx, _cache) {
+          const _directive_awesome_directive = _resolveDirective("awesome-directive")
+          const _directive_named_directive = _resolveDirective("named-directive")
+
+          return _withDirectives((_openBlock(), _createElementBlock("div", {
+            onClick: _cache[0] || (_cache[0] = (...args) => (_ctx.foo && _ctx.foo(...args)))
+          }, null, 512 /* NEED_PATCH */)), [
+            [_directive_awesome_directive],
+            [_directive_named_directive]
+          ])
+        }"
+      `)
+      expect(replaceRoot((await ctx.injectImports(defaultDirective.code, 'a.vue')).code.toString())).toMatchInlineSnapshot(`
+        "import { NamedDirective as _directive_named_directive } from '/directives/named-directive.ts';
+        import _directive_awesome_directive from '/directives/awesome-directive.ts';import { withDirectives as _withDirectives, openBlock as _openBlock, createElementBlock as _createElementBlock } from "vue"
+
+        export function render(_ctx, _cache) {
+          return _withDirectives((_openBlock(), _createElementBlock("div", {
+            onClick: _cache[0] || (_cache[0] = (...args) => (_ctx.foo && _ctx.foo(...args)))
+          }, null, 512 /* NEED_PATCH */)), [
+            [_directive_awesome_directive],
+            [_directive_named_directive]
+          ])
+        }"
+      `)
+    })
+
+    it('dts', async () => {
+      expect(replaceRoot(await ctx.generateTypeDeclarations())).toMatchInlineSnapshot(`
+        "export {}
+        declare global {
+          const NamedDirective: typeof import('/directives/named-directive')['NamedDirective']
+          const default: typeof import('/directives/awesome-directive')['default']
```

Comment:
This should be `vAwesomeDirective`, I also tried running `await ctx.init()` with same result: in the playground this global is fine